index.js

```diff
@@ -102,6 +106,18 @@ AFRAME.registerComponent('teleport-controls', {
     this.hitEntity.setAttribute('visible', false);
 
     if ('collisionEntities' in diff) { this.queryCollisionEntities(); }
+
+    if (oldData.aimEvent !== data.aimEvent) {
+      el.removeEventListener(oldData.aimEvent, this.onAim);
+      el.addEventListener(data.aimEvent, this.onAim);
+    }
+
+    if (oldData.teleportEvent !== data.teleportEvent) {
+      el.removeEventListener(oldData.teleportEvent, this.onTeleport);
+      el.addEventListener(data.teleportEvent, this.onTeleport);
+    }
+    el.addEventListener(data.teleportEvent, this.onTeleport);
```

Comment:
I think this line is not needed.

Reply:
@vincentfretin yep, nice catch! thanks :D In any case don't take this PR too serious yet as it's a work in progress and it will should change a lot before being ready to merge

Reply:
Sure. I'm following now closely the components I use to stay alert on the changes. I don't mind reviewing your commits if you want to, even if it's a wip.